### PR TITLE
feat(hc): Set up region resolution on RPC services

### PR DESCRIPTION
**Pull request by RyanSkonnord**

### CODE REVIEW

**Comment by RyanSkonnord** [comment]
Previously reviewed at https://github.com/getsentry/sentry/pull/45448

File: src/sentry/services/hybrid_cloud/organization/__init__.py [hunk 1]
```diff
@@ -188,7 +195,7 @@ def get_organization_by_id(
 
     # TODO: This should return RpcOrganizationSummary objects, since we cannot realistically span out requests and
     #  capture full org objects / teams / permissions.  But we can gather basic summary data from the control silo.
-    @rpc_method
+    @regional_rpc_method(resolve=UnimplementedRegionResolution())
```
Comment (corps) [inline src/sentry/services/hybrid_cloud/organization/__init__.py:198]: 👍

**Review by corps — APPROVED** [review]

File: src/sentry/services/hybrid_cloud/region.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    def organization_mapping_manager(self) -> BaseManager[OrganizationMapping]:
+        from sentry.models import OrganizationMapping
+
+        # Convenience method to avoid repeating the local import
+        return OrganizationMapping.objects
+
+
+@dataclass(frozen=True)
+class ByOrganizationObject(RegionResolution):
+    """Resolve from a parameter representing an organization object."""
```
Comment (RyanSkonnord) [inline src/sentry/services/hybrid_cloud/region.py:38]: Following up on [this comment](https://github.com/getsentry/sentry/pull/45448#discussion_r1128536536) from @markstory, now that this is done, the case where this object finds the right parameter by inference didn't come up as often I was expecting. The inference code plainly isn't worth it, so I got rid of it. (https://github.com/getsentry/sentry/pull/46638/commits/6a1b698c1d2fbaa0894cabe03d9343bb37bd798c)

**Review by markstory — APPROVED** [review]
Nice to see that the region selector has gotten simpler :clap:

**Review by dashed — APPROVED** [review]

File: src/sentry/services/hybrid_cloud/organization/__init__.py [hunk 3]
```diff
@@ -264,22 +271,22 @@ def add_organization_member(
...
-    @rpc_method
+    @regional_rpc_method(resolve=ByOrganizationIdAttribute("organization_member"))
     @abstractmethod
     def add_team_member(self, *, team_id: int, organization_member: RpcOrganizationMember) -> None:
         pass
 
-    @rpc_method
+    @regional_rpc_method(resolve=UnimplementedRegionResolution())
     @abstractmethod
     def get_team_members(self, *, team_id: int) -> Iterable[RpcOrganizationMember]:
```
[the remark this answers is not in the document]
Comment (RyanSkonnord) [reply src/sentry/services/hybrid_cloud/organization/__init__.py:286]: You mean right now, or eventually in prod?

Right now, it's the same behavior as when `RpcService` raises an `RpcServiceUnimplementedException`. We log the stopping point in the remote flow and then fall back to the local database-backed implementation. We return the result as though it happened remotely and continue on our way.

In prod, an exception or null result here would generally just throw an error. There may be cases that need special handling (`OrganizationService.get_organizations` jumps out at me) that we could implement by one-off `RegionResolution` classes. But for now I'm operating under the assumption that any call to a `local_mode = SiloMode.REGION` service, from the control silo, must resolve to one and only one region: if we run into "no region" or "multiple regions" cases that still need to return sensible results, we'll have to add code in `RpcService` to accommodate that.

**Pull request merged**